lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -450,12 +450,18 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
     StringRef Dir;
     StringRef File;
     SmallString<128> DirBuf;
-    SmallString<128> FileBuf;
-    std::string RemappedFileString = DebugPrefixMap.remapPath(FileName);
-    SmallString<128> RemappedFile = StringRef(RemappedFileString);
-    llvm::sys::path::remove_dots(RemappedFile);
+    SmallString<128> FileBuf(FileName);
+    llvm::sys::path::remove_dots(FileBuf);
+    std::string RemappedFileString = DebugPrefixMap.remapPath(FileBuf);
+    FileBuf.clear();
```

Comment:
Instead of using `FileBuf` for two purposes (and needing this clear) what do you think of having separate vars?

Reply:
I wasn't sure if stack coloring was good enough detect that the memory can be reused and stack space comes at a premium in the compiler. That said this is pretty much a leaf function, so it probably wouldn't much.